File: ChoreAssignmentTracker/src/FamilyMembers.Core/Models/FamilyMember.cs
namespace FamilyMembers.Core.Models;

public class FamilyMember
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public int Age { get; set; }
    public string? Role { get; set; }
    public int TotalPoints { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
    public Guid? TenantId { get; set; }
}

File: ChoreAssignmentTracker/src/ChoreAssignmentTracker.Web/projects/api/src/lib/models/family-member.ts
export interface FamilyMember {
  familyMemberId: string;
  userId: string;
  name: string;
  age?: number;
  avatar?: string;
  totalPoints: number;
  availablePoints: number;
  isActive: boolean;
  createdAt: Date;
  completionRate: number;
}

export interface CreateFamilyMember {
  userId: string;
  name: string;
  age?: number;
  avatar?: string;
}

export interface UpdateFamilyMember {
  name: string;
  age?: number;
  avatar?: string;
  isActive: boolean;
}
